src/main/java/com/google/sps/utility/FreeTimeUtility.java

```diff
@@ -21,154 +21,122 @@
 /** Class containing the user's free hours. */
 public final class FreeTimeUtility {
 
-  private final long startDay;
-  private final List<Long> workHoursPerDay;
-  private final List<Long> personalHoursPerDay;
-  private final long numMillisecondsDay = TimeUnit.HOURS.toMillis(24);
-  private final long Hour = TimeUnit.HOURS.toMillis(1);
-  private final int UnixEpochDayShift = 3;
-  private final long personalBegin = 7 * Hour;
-  private final long workBegin = 10 * Hour;
-  private final long workEnd = 18 * Hour;
-  private final long personalEnd = 23 * Hour;
+  private final Date startDate;
+  private List<Long> workHoursPerDay;
+  private List<Long> personalHoursPerDay;
+  private List<DatePair> morningFreeInterval;
+  private List<DatePair> workFreeInterval;
+  private List<DatePair> eveningFreeInterval;
+  private static final long HOUR_MILLI = TimeUnit.HOURS.toMillis(1);
+  private static final long DAY_MILLI = TimeUnit.DAYS.toMillis(1);
+  private static final long PERSONAL_BEGIN = 7 * HOUR_MILLI;
+  private static final long WORK_BEGIN = 10 * HOUR_MILLI;
+  private static final long WORK_END = 18 * HOUR_MILLI;
+  private static final long PERSONAL_END = 23 * HOUR_MILLI;
+  private static final int OFFSET = 7;
 
   /**
-   * Initialize the class by calculating the start day. The free hours should be eight hours for the
-   * working hours and for the pesonal time initially. Adjust the free hours on the first day to
-   * account for the fact that there are hours that have already passed
+   * Initialize the class with the start day. The work hours are harrrd-coded between 10 AM and 6
+   * PM. The rest of the free hours are between 7 AM and 11 PM.
    *
-   * @param startTime parameter of type long that gives the Unix epoch time of the start/now. The
-   *     timezone is UTC
+   * @param startTime parameter that gives the time of the start/now.
    */
-  public FreeTimeUtility(long startTime) {
-    long numDays = startTime / numMillisecondsDay;
-    this.startDay = (numDays + UnixEpochDayShift) % 7;
-    long eightHours = 8 * Hour;
-    this.workHoursPerDay =
-        Arrays.asList(eightHours, eightHours, eightHours, eightHours, eightHours);
-    this.personalHoursPerDay =
-        Arrays.asList(eightHours, eightHours, eightHours, eightHours, eightHours);
-    setEndDayEvent(0, startTime % numMillisecondsDay);
+  public FreeTimeUtility(Date startDate) {
+    this.startDate = startDate;
+    Date basisDate = new Date(startDate.getYear(), startDate.getMonth(), startDate.getDate(), 0, 0);
+
+    this.morningFreeInterval = new ArrayList<>();
+    this.workFreeInterval = new ArrayList<>();
+    this.eveningFreeInterval = new ArrayList<>();
+
+    for (int day = 0; day < 5; day++) {
+      Date workStart = new Date(basisDate.getTime() + day * DAY_MILLI + WORK_BEGIN);
```

Comment:
Nit: [Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) could make this slightly simpler. Example: https://github.com/googleinterns/step7-2020/blob/72762882bf046a75fd8c8ff2c6a7f03844277c15/src/main/java/com/google/sps/servlets/TasksServlet.java#L192

Reply:
Used this. `Resolved`